Source/ASTextNode2.mm

```diff
@@ -313,6 +330,102 @@ - (UIAccessibilityTraits)defaultAccessibilityTraits
   return UIAccessibilityTraitStaticText;
 }
 
+static void ASUpdateAccessibilityFrame(ASTextNodeAccessiblityElement *accessibilityElement, ASTextLayout *layout, ASDisplayNode * _Nullable containerNode, ASDisplayNode *node) {
+  containerNode = containerNode ?:  ASFirstNonLayerBackedSupernodeForNode(node);
+  CGRect textLayoutFrame = CGRectZero;
+  if (accessibilityElement.accessibilityRange.location == NSNotFound) {
+    // If no accessibilityRange was specified (as is done for the text element), just use the label's frame.
+    textLayoutFrame = [layout rectForRange:[ASTextRange rangeWithRange:NSMakeRange(0, accessibilityElement.accessibilityLabel.length)]];
+    // Handle text is too large for the layout's text container size
+    if (textLayoutFrame.origin.x == INFINITY) {
+      textLayoutFrame = [layout rectForRange:[ASTextRange rangeWithRange:layout.visibleRange]];
+    }
+  } else {
+    textLayoutFrame = [layout rectForRange:[ASTextRange rangeWithRange:accessibilityElement.accessibilityRange]];
+  }
+  CGRect accessibilityFrame = [containerNode convertRect:textLayoutFrame fromNode:node];
+  accessibilityElement.accessibilityFrame = UIAccessibilityConvertFrameToScreenCoordinates(accessibilityFrame, containerNode.view);
+}
+
+/// Walks up the node tree and searches for the first node that is not layer backed
+static ASDisplayNode *ASFirstNonLayerBackedSupernodeForNode(ASDisplayNode *node) {
+  ASDisplayNode *containerNode = node;
+  while (containerNode.isLayerBacked) {
+    containerNode = containerNode.supernode;
+  }
+  return containerNode;
+}
```

Comment:
We can keep this but we also already have ASFindClosestViewOfLayer. Since we're on the main thread, traversing the layer hierarchy might make sense.

Reply:
We should not touch the view in here as we are also calling it within a layer backed node.

Reply:
`ASFindClosestViewOfLayer` doesn't touch the view. It finds a view for a given layer.